client/rpc/trace/src/lib.rs

```diff
@@ -273,32 +144,42 @@ where
 
 			let block_hash = block_header.hash();
 
-			let cache_block = match cached_blocks.entry(block_hash) {
-				Entry::Occupied(entry) => {
-					tracing::trace!(block_height, %block_hash, "Cache hit, no need to replay block !");
-
-					let cache_block = entry.into_mut();
-					cache_block.expiration = Instant::now() + cache_duration;
+			block_hashes.push(block_hash);
+		}
 
-					cache_block
-				}
-				Entry::Vacant(entry) => {
-					tracing::trace!(block_height, %block_hash, "Cache miss, replaying block ...");
+		// Start a batch with these blocks.
+		let batch_id = self.requester.start_batch(block_hashes.clone()).await?;
+		// Fetch all the traces. It is done in another function to simplify error handling and allow
+		// to call the following `stop_batch` regardless of the result. This is important for the
+		// cache cleanup to work properly.
+		let res = self.fetch_traces(req, &block_hashes, count as usize).await;
```

Comment:
missing `?`

Reply:
As explained in the comment, we want to call `self.requester.stop_batch` in all cases, even if there is an error.
Also the type of `res` here matches the type of this function (filter), so `?` would not work.